python/chemiscope/adapters.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+try:
+    import ase
+
+    HAVE_ASE = True
+except ImportError:
+    HAVE_ASE = False
+
+
+def frames_to_json(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return [_ase_to_json(frame) for frame in frames]
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def atom_properties(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return [_ase_atom_properties(frame) for frame in frames]
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def structure_properties(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return [_ase_structure_properties(frame) for frame in frames]
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def _ase_to_json(frame):
+    data = {}
+    data["size"] = len(frame)
+    data["names"] = list(frame.symbols)
+    data["x"] = [float(value) for value in frame.positions[:, 0]]
+    data["y"] = [float(value) for value in frame.positions[:, 1]]
+    data["z"] = [float(value) for value in frame.positions[:, 2]]
+
+    if (frame.cell.lengths() != [0.0, 0.0, 0.0]).all():
+        data["cell"] = list(np.concatenate(frame.cell))
+
+    return data
+
+
+IGNORED_ASE_ARRAYS = ["positions", "numbers"]
+
+
+def _ase_atom_properties(frames):
+    properties = {
+        name: {"target": "atom", "values": value}
+        for name, value in frames[0].arrays.items()
+        if name not in IGNORED_ASE_ARRAYS
+    }
+
+    for frame in frames[1:]:
+        for name, value in frame.arrays.items():
+            if name in IGNORED_ASE_ARRAYS:
+                continue
+            properties[name]["values"] = np.concatenate(
+                [properties[name]["values"], value]
+            )
+    return properties
```

Comment:
I don't love this instantiation. If there are frames which don't have a full set of properties, then we'll end up with later frames' properties being associated with the wrong frame. Also causing an error: frames with a property that doesn't exist in previous frames. How about:

```suggestion
    properties = {
        name: {"target": "atom", "values": value}
        for name, value in frames[0].arrays.items()
        if name not in IGNORED_ASE_ARRAYS
    }

    for frame in frames[1:]:
        for name, value in frame.arrays.items():
            if name in IGNORED_ASE_ARRAYS:
                continue
            if name not in properties:
                names = list(properties.keys())
                properties[name] = {"target": "atom",  "values": np.nan * np.zeros(properties[names[-1]]['values'].shape)}
        for name in properties.keys():
            value = frame.arrays.get(name, np.nan * np.zeros(len(frame.positions)))
            properties[name]["values"] = np.concatenate(
                [properties[name]["values"], value]
            )
            
    return properties
```

Reply:
Make sense, I'll change it!

Reply:
Done in 73d60deb65a9a778e1be9b5dea35af7fac3f8f11